noaa/gefs/forecast/reformat.py

```diff
@@ -646,3 +657,67 @@ def consume[T](iterator: Iterable[T], n: int | None = None) -> None:
         deque(iterator, maxlen=0)
     else:
         next(islice(iterator, n, n), None)
+
+
+def check_recent_nans(
+    ds: xr.Dataset, n_samples: int = 100, max_nan_percentage: float = 70
+) -> validation.ValidationResult:
+    """Check for NaN values in the most recent day of data. Fails if more than 70% of sampled data is NaN."""
```

Comment:
The last day's data is probably going to be mostly NaNs past the first 10 days in the `lead_time` dimension, except for a handful of ensemble members which get filled in quicker.
I wonder if a more tailored approach would make more sense here.
1) Change this to `sel` down to 10 days and require a lower `max_nan_percentage` for the `latest_init`
2) Check that the newest `init_time` that's more than 3 days old has a low % of nans in all lead times.